Check powerstate of switch

wopfel opened the pull request:
You can check the powerstate of the switch. Pass `--powerstate on` as  command line argument, and the script will give you a warning state when  the Shelly switch is powered off.

This PR relates to #2.

Some notes about the commit:

- I'm not very familiar using Python. Hopefully you're not getting mad when you see my code.
- `exit_state` is set to 1 without any further checks (this will break things when there's a check for a critical state in the future)

I wanted to share the code early. 

Feedback much appreciated.

wopfel committed:
Check powerstate of switch

You can check the powerstate of the switch. Pass `--powerstate on` as 
command line argument, and the script will give you a warning state when 
the Shelly switch is powered off.

Napsty commented:
> Hopefully you're not getting mad when you see my code. 

I personally would have used booleans (True/False) but this also works just fine. :+1:

Just one more thing before I merge, could you please test all the check types? Just to make sure that the new `powerstate` variable does not cause an issue on the other checks. As mentioned, I am unable to test here locally. Thanks!

wopfel commented:
Thank you very much for your feedback.

> I personally would have used booleans (True/False) but this also works just fine. 👍

I'm not very happy using strings here, too. So could you give me a hint how to do it the boolean way?

https://github.com/Napsty/check_shelly/blob/224c487ef3259ff40c359f0eba4a02ce852f5d78/check_shelly.py#L197

To be honest, I'm also not very happy with the names, `powerstate` and `powerstatus`. They're very similar and without looking into the code, I cannot tell which one is the args-related variable and which one is the variable representing the device. Well, it works that way, but it's a bit strange. Any ideas? 


> Just one more thing before I merge, could you please test all the check types? Just to make sure that the new `powerstate` variable does not cause an issue on the other checks. As mentioned, I am unable to test here locally. Thanks!

Seems good.

```
# python3 /usr/local/bin/check_shelly.py -H 192.168.14.121 -t info ; echo $?
SHELLY OK: Device Plus1PM (Model: SNSW-001P16EU, Generation: 2, Firmware: 0.12.0) is running - Authentication is disabled
0

# python3 /usr/local/bin/check_shelly.py -H 192.168.14.121 -t system ; echo $?
SHELLY OK: Device (shellyplus1pm-xxx), uptime 414025 |uptime=414025 memory=107712B;;;0;249160 disk=348160B;;;0;458752
0

# python3 /usr/local/bin/check_shelly.py -H 192.168.14.121 -t meter ; echo $?
SHELLY OK: Device (shellyplus1pm-xxx) SWITCH_0 is off, currently using 0 Watt / 0 Amp |power=0 current=0 total_power=50.983 temp=33.7 powerstatus=0
0
```

Napsty commented:
> To be honest, I'm also not very happy with the names, powerstate and powerstatus.

Maybe use `--expect-power` as parameter, just as an idea? It's up to you of course. That would result in:

```
./check_shelly.py -H IPofShelly -t meter -a -p secret -s 1 --expect-power on
```

wopfel committed:
Rename variables to powerstatus

The variables were named powerstate, and powerstatus before. This was 
misleading, I think. Since they're all about the same topic, the  
variables are named powerstatus now.

The --powerstate argument was renamed to --expect-powerstatus.

wopfel committed:
Boolean check of expect_powerstatus

wopfel committed:
Add comment

wopfel committed:
Exit if checktype!=meter but expect-powerstatus is given

wopfel commented:
Thanks for the inspiration. I call it `--expect-powerstatus` now. The variables are named "powerstatus" everywhere.

I've also changed the if-clause. You've mentioned you'd have used booleans. The code looks much better now I think.

What do you think?

Napsty commented:
LGTM! Thanks!
